src/api-service/__app__/onefuzzlib/workers/scalesets.py

```diff
@@ -449,6 +449,29 @@ def _resize_shrink(self, to_remove: int) -> None:
         for node in nodes:
             node.send_stop_if_free()
 
+    def check_scaleset_size(self) -> None:
```

Comment:
Can we add a doc comment summarizing the responsibilities of this function?

Also, I find the name a bit confusing (`check` makes me think of a read-only predicate or guard function, but this does a lot of mutating). How about something like `sync_scaleset_size()`?